src/main/java/datawave/microservice/authorization/federation/FederatedAuthorizationService.java

```diff
@@ -0,0 +1,197 @@
+                                    + currentUser.getPrimaryUser() + ", Status Code: " + authorizationsListBaseResponseEntity.getStatusCodeValue());
+                }
+            } else {
+                authorizationException = new AuthorizationException(
+                                "Unknown error occurred while calling listEffectiveAuthorizations for " + currentUser.getPrimaryUser());
+            }
+            throw authorizationException;
+        } catch (ServiceException e) {
+            log.error("Timed out waiting for federated listEffectiveAuthorizations response");
+            throw new AuthorizationException("Timed out waiting for federated listEffectiveAuthorizations response", e);
+        }
+    }
+    
+    @Override
+    public GenericResponse<String> flushCachedCredentials(ProxiedUserDetails currentUser) throws AuthorizationException {
+        return flushCachedCredentials(currentUser, true);
+    }
+    
+    public GenericResponse<String> flushCachedCredentials(ProxiedUserDetails currentUser, boolean federate) throws AuthorizationException {
+        log.info("FederatedAuthorizationService flushCachedCredentials (federate: {}) for {}", federate, currentUser.getPrimaryUser());
+        
+        RetryTimeoutProperties retry = federatedAuthorizationProperties.getFlushCachedCredentials();
+        try {
+            // @formatter:off
+            ResponseEntity<GenericResponse> genericResponseEntity = webClient.get()
+                    .uri(uriBuilder -> uriBuilder
+                            .path("flushCachedCredentials")
+                            .queryParam("includeRemoteServices", federate)
+                            .build())
+                    .header(ENTITIES_HEADER, getProxiedEntities(currentUser))
+                    .header(ISSUERS_HEADER, getProxiedIssuers(currentUser))
+                    .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+                    .retrieve()
+                    // don't retry on 4xx errors
+                    .onStatus(HttpStatus::is5xxServerError,
+                            clientResponse -> Mono.error(new ServiceException("Service Error", clientResponse.rawStatusCode())))
+                    .toEntity(GenericResponse.class)
+                    .retryWhen(Retry
+                            .fixedDelay(retry.getRetries(), Duration.ofMillis(retry.getRetryDelayMillis()))
+                            .filter(throwable -> throwable instanceof ServiceException)
+                            .onRetryExhaustedThrow(((retryBackoffSpec, retrySignal) -> {
+                                throw new ServiceException("External Service failed to process after max retries",
+                                        HttpStatus.SERVICE_UNAVAILABLE.value());
+                            })))
+                    .block(Duration.ofMillis(retry.getTimeoutMillis()));
+            // @formatter:on
+            
+            AuthorizationException authorizationException;
```

Comment:
Similarly, it looks like the only difference between this block of code and the other is the name of the action, which could perhaps be a parameter to a new function.

Reply:
should be fixed with the latest push.